### PR TITLE
Made Item entities hurtable by fire or cacti

### LINKED ISSUE #571: Items that are not netherite don't burn in fire or lava

### PR DESCRIPTION
**Pull request by JustTalDevelops**
Resolves #571.

### CODE REVIEW

File: server/entity/item.go [hunk 1]
```diff
@@ -70,6 +82,113 @@ func (it *Item) SetPickupDelay(d time.Duration) {
...
+	case armourTiered:
+		if _, ok := i.ArmourTier().(item.ArmourTierNetherite); ok {
+			return false
+		}
+	case toolTiered:
+		if i.ToolTier() == item.ToolTierNetherite {
+			return false
+		}
+	}
+	return true
```
Comment (DaPigGuy) [inline server/entity/item.go:189]: This logic seems flipped?

File: server/entity/item.go [hunk 2]
```diff
@@ -70,6 +82,113 @@ func (it *Item) SetPickupDelay(d time.Duration) {
...
+type fireSource interface {
+	Fire() bool
+}
+
+// Hurt ...
+func (it *Item) Hurt(dmg float64, source damage.Source) (n float64, vulnerable bool) {
+	if dmg < 0 {
+		return 0, true
+	}
+	if f, ok := source.(fireSource); ok && f.Fire() && !it.FireProof() {
```
Comment (DaPigGuy) [inline server/entity/item.go:120]:
```suggestion
	if f, ok := source.(fireSource); ok && f.Fire() && it.FireProof() {
```

File: server/entity/item.go (hunk 1 above)
Comment (DaPigGuy) [inline server/entity/item.go:189]:
```suggestion
	switch i := it.Item().Item().(type) {
	case armourTiered:
		if _, ok := i.ArmourTier().(item.ArmourTierNetherite); ok {
			return true
		}
	case toolTiered:
		return i.ToolTier() == item.ToolTierNetherite
	}
	return false
```

File: server/entity/item.go [hunk 3]
```diff
@@ -70,6 +82,113 @@ func (it *Item) SetPickupDelay(d time.Duration) {
...
+// OnFireDuration ...
+func (it *Item) OnFireDuration() time.Duration {
+	it.mu.Lock()
+	defer it.mu.Unlock()
+	return time.Duration(it.fireTicks) * time.Second / 20
+}
+
+// SetOnFire ...
+func (it *Item) SetOnFire(duration time.Duration) {
+	if !it.FireProof() {
```
Comment (DaPigGuy) [inline server/entity/item.go:148]:
```suggestion
	if it.FireProof() {
```

**Comment by T14Raptor** [comment]
Closing due to the entity system rewrite

**Pull request closed without merging**